source/simulator/assemblers/stokes.cc

```diff
@@ -264,6 +266,18 @@ namespace aspect
     {
       const unsigned int n_points = outputs.viscosities.size();
 
+      if (this->get_parameters().enable_elasticity
+          && outputs.template get_additional_output<MaterialModel::AdditionalMaterialOutputsStokesRHS<dim> >() == NULL)
+        {
+          outputs.additional_outputs.push_back(
+            std_cxx11::shared_ptr<MaterialModel::AdditionalMaterialOutputsStokesRHS<dim> >
+            (new MaterialModel::AdditionalMaterialOutputsStokesRHS<dim> (n_points)));
+        }
+      Assert(!this->get_parameters().enable_elasticity
+             ||
+             outputs.template get_additional_output<MaterialModel::AdditionalMaterialOutputsStokesRHS<dim> >()->rhs_e.size()
+             == n_points, ExcInternalError());
```

Comment:
You can combine all of your new lines with the ones below for the right-hand side, as you do exactly the same things for elasticity as for additional_stokes_rhs. So no need to copy all of that, just combine the if-statements and the Asserts.